### DISCUSSION TITLE
Not receiving Error from Send API

### DISCUSSION THREAD

**Discussion by dadams39** [microsoftgraph/msgraph-sdk-go#166] [General]
Related to  this [question](https://docs.microsoft.com/en-us/answers/questions/856710/microsoft-graph-email-failures.html)

When the message `sender` does not match the `user`, a message will fail to send and the user will receive an e-mail. 
Code:
```go
result , err := client.UsersById(user).MailFoldersById(mailFolderId).Messages().Post(requestBody) 
```
This is problematic. This is also true if there is a failure and the application sends 4 of the same message. The 3 remaining messages will be caught by the spam filter and the API generates neither error nor any additional information stating that there was potentially a problem.

**Comment by baywet** [comment]
but the API call was successful, correct? (200 range response) and the SDK correctly serialized and sent the payload?
The focus of this repository (and of the issues/discussions) is really about the SDK, and not about how individual APIs behave.
The best thing to do when talking about API behaviour, is to reach out [on the dedicated Microsoft Q&A forum](https://docs.microsoft.com/en-us/answers/topics/microsoft-graph-mail.html) which the service team is watching, you'll get better answers this way.

**Reply by dadams39** [reply]
I have contacted the Q&A forum for other aspects of this. But consider sending the same message 3 times ( which is something not to do), the call is successful. However, it is NOT successful. It was picked up by the spam folder. Do you believe the SDK should receive feedback in such cases?

**Reply by baywet** [reply]
The SDK only gets the information the API provides it. It's not up to us (the SDK team) to make decisions about how this or that API under Microsoft Graph should behave. It's up to the API owners (helped by a review board to make sure consumers have a unified experience).

Then there is the technical challenge of your specific ask. Whenever you send an email through Microsoft Graph, it relies on exchange transport if the email is going to the same organization you're sending things from, so in that case we could imagine implementing live feedback to the API consumer. But in the case it's going to any other organization, it relies on SMTP. And SMTP doesn't provide feedback on what it does with an email once it's been received.
